### PR TITLE
Disable Ctrl-C handling on WASM

### PR DESCRIPTION
**Pull request by bjorn3**
WASM fundamentally doesn't support signals. If WASI ever gets support for notifying the guest process of a Ctrl-C that happened, this would have to be done through the guest process polling for the signal, which will require thread support in WASI too to be compatible with the api provided by the ctrlc crate.

### CODE REVIEW

**Comment by lcnr** [comment]
this seems reasonable to me, I don't know much about wasm though :shrug: cc @alexcrichton maybe? What's the impact of simply using `ctrlc` in wasm? do you get compilation failure/linker errors or is it simply a noop?

r=me after CI

**Comment by bjorn3** [comment]
CI failure is because of a tidy bug. It checks if a target dependent dependency applies to wasm and if so ignores it. This check however also triggers if it doesn't just apply to wasm but also to other targets. So for example `cfg(not(windows))` would also be ignored as wasm is matched by `cfg(not(windows))`. Or in this case `cfg(not(target_family = "wasm"))]` is considered to be matched by wasm as `target_family = "wasm"` is not explicitly set in the check. I've pushed a commit to remove the filtering entirely. It only filtered out two deps anyway. At least one of which is incorrect and I think the other one is also incorrect.

**Comment by bjorn3** [comment]
> What's the impact of simply using ctrlc in wasm? do you get compilation failure/linker errors or is it simply a noop?

I got a compilation failure when compiling the ctrlc crate.

<details><summary>The errors in question</summary>

```
error[E0432]: unresolved import `platform::Signal`
  --> /home/gh-bjorn3/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ctrlc-3.4.4/src/lib.rs:53:9
   |
53 | pub use platform::Signal;
   |         ^^^^^^^^^^^^^^^^ no `Signal` in `platform`

error[E0412]: cannot find type `Error` in module `platform`
  --> /home/gh-bjorn3/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ctrlc-3.4.4/src/error.rs:25:21
   |
25 | impl From<platform::Error> for Error {
   |                     ^^^^^ not found in `platform`
   |
help: consider importing one of these items
   |
1  + use core::error::Error;
   |
1  + use core::fmt::Error;
   |
1  + use crate::Error;
   |
1  + use crate::error::fmt::Error;
   |
     and 3 other candidates
help: if you import `Error`, refer to it directly
   |
25 - impl From<platform::Error> for Error {
25 + impl From<Error> for Error {
   |

error[E0412]: cannot find type `Error` in module `platform`
  --> /home/gh-bjorn3/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ctrlc-3.4.4/src/error.rs:26:26
   |
26 |     fn from(e: platform::Error) -> Error {
   |                          ^^^^^ not found in `platform`
   |
help: consider importing one of these items
   |
1  + use core::error::Error;
   |
1  + use core::fmt::Error;
   |
1  + use crate::Error;
   |
1  + use crate::error::fmt::Error;
   |
     and 3 other candidates
help: if you import `Error`, refer to it directly
   |
26 -     fn from(e: platform::Error) -> Error {
26 +     fn from(e: Error) -> Error {
   |

error[E0433]: failed to resolve: could not find `Error` in `platform`
  --> /home/gh-bjorn3/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ctrlc-3.4.4/src/error.rs:28:27
   |
28 |         if e == platform::Error::EEXIST {
   |                           ^^^^^ could not find `Error` in `platform`
   |
help: consider importing one of these items
   |
1  + use core::error::Error;
   |
1  + use core::fmt::Error;
   |
1  + use crate::Error;
   |
1  + use crate::error::fmt::Error;
   |
     and 3 other candidates
help: if you import `Error`, refer to it directly
   |
28 -         if e == platform::Error::EEXIST {
28 +         if e == Error::EEXIST {
   |

error[E0412]: cannot find type `Signal` in module `platform`
  --> /home/gh-bjorn3/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ctrlc-3.4.4/src/signal.rs:22:21
   |
22 |     Other(platform::Signal),
   |                     ^^^^^^ not found in `platform`

error[E0425]: cannot find function `init_os_handler` in module `platform`
   --> /home/gh-bjorn3/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ctrlc-3.4.4/src/lib.rs:134:25
    |
134 |         match platform::init_os_handler(overwrite) {
    |                         ^^^^^^^^^^^^^^^ not found in `platform`

error[E0425]: cannot find function `block_ctrl_c` in module `platform`
   --> /home/gh-bjorn3/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ctrlc-3.4.4/src/lib.rs:146:27
    |
146 |                 platform::block_ctrl_c().expect("Critical system error while waiting for Ctrl-C");
    |                           ^^^^^^^^^^^^ not found in `platform`

Some errors have detailed explanations: E0412, E0425, E0432, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `ctrlc` (lib) due to 7 previous errors
```

</details>

File: src/tools/tidy/src/deps.rs [hunk 1]
```diff
@@ -668,27 +670,7 @@ fn check_permitted_dependencies(
     let mut deps = HashSet::new();
     for to_check in restricted_dependency_crates {
         let to_check = pkg_from_name(metadata, to_check);
-        use cargo_platform::Cfg;
-        use std::str::FromStr;
-        // We don't expect the compiler to ever run on wasm32, so strip
```
Comment (Nilstrieb) [inline src/tools/tidy/src/deps.rs:673]: 😈

**Review by Noratrieb — APPROVED** [review]
r=me

**Comment by klensy** [comment]
`cargo-platform` became dead dep?

**Comment by alexcrichton** [comment]
Looks good to me as well 👍

**Comment by bjorn3** [comment]
Yes, the cargo-platform dependency can be removed. I've opened https://github.com/rust-lang/rust/pull/123796 for this as this PR is already being tested by bors.

**Pull request merged**